modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/create_folder_command.rb

```diff
@@ -33,24 +33,31 @@ module Peripherals
     module StorageInteraction
       module OneDrive
         class CreateFolderCommand
-          def self.call(storage:, folder_path:)
-            new(storage).call(folder_path:)
+          using ServiceResultRefinements
+
+          def self.call(storage:, folder_path:, parent_location: nil)
```

Comment:
🔴 this interface is now no longer similar to the nextcloud command.

I understand, why this is needed: OneDrive cannot derive the parent location from the path of the folder to be created. But the solution would be to just add this to the interface of the nextcloud command.

Reply:
My first intention would be to adapt NC command to also eat this parameter. Maybe even making it non optional. Then you would read the signature as: "you need the storage, where you wanna create - a location identifier - and a parent context, given as a location identifier". For OneDrive we need it all to create any folder anywhere. (useful also for the soon to be implemented feature of creating folders from the location picker).

For nextcloud we basically can ignore the last argument or even interpret it like:

```ruby
CreateFolderCommand.call(storage:, folder_location: '/Folder', parent_location: '/Parent')
```
is similar in the resulting execution to
```ruby
CreateFolderCommand.call(storage:, folder_location: '/Parent/Folder', parent_location: nil)
CreateFolderCommand.call(storage:, folder_location: '/Parent/Folder', parent_location: '/')
CreateFolderCommand.call(storage:, folder_location: '/Parent/Folder', parent_location: '')
```

Reply:
You understand my sentiment here?

Reply:
Yeah, then lets do that. This PR should no be the place for more wide range refactoring. Instead, I'd really like to keep the interface in common, so that this thingies do not evolve into sleepers.